lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -677,26 +680,64 @@ static LogicalResult hasRequiredPorts(ComponentOp op) {
          << difference;
 }
 
+// A helper function to get the number of invoke operations in a block.
+size_t getInvokeOpNumber(Block *block) {
+  if (!block)
+    return 0;
+
+  auto iIt = block->getOps<InvokeOp>();
```

Comment:
This is the second time I've seen code of this sort. Maybe have some kind of helper functions so that we don't need to update two places if an `InvokeOp` can live in an additional body?